source/shared/cpp/ObjectModel/SharedAdaptiveCard.cpp

```diff
@@ -110,7 +91,32 @@ std::shared_ptr<ParseResult> AdaptiveCard::Deserialize(const Json::Value& json,
         warnings.push_back(std::make_shared<AdaptiveCardParseWarning>(AdaptiveSharedNamespace::WarningStatusCode::InvalidLanguage,
                                                                       "Invalid language identifier: " + language));
     }
+}
 
+#ifdef __ANDROID__
+std::shared_ptr<ParseResult> AdaptiveCard::Deserialize(const Json::Value& json,
+                                                       std::string rendererVersion,
+                                                       ParseContext& context) throw(AdaptiveSharedNamespace::AdaptiveCardParseException)
+#else
+std::shared_ptr<ParseResult> AdaptiveCard::Deserialize(const Json::Value& json, std::string rendererVersion, ParseContext& context)
+#endif // __ANDROID__
+{
+    ParseUtil::ThrowIfNotJsonObject(json);
+
+    const bool enforceVersion = !rendererVersion.empty();
+
+    // Verify this is an adaptive card
+    ParseUtil::ExpectTypeString(json, CardElementType::AdaptiveCard);
+
+    std::string version = ParseUtil::GetString(json, AdaptiveCardSchemaKey::Version, enforceVersion);
+    std::string fallbackText = ParseUtil::GetString(json, AdaptiveCardSchemaKey::FallbackText);
+    std::string language = ParseUtil::GetString(json, AdaptiveCardSchemaKey::Language);
+    std::string speak = ParseUtil::GetString(json, AdaptiveCardSchemaKey::Speak);
+
+    // check if language is valid
+    _ValidateLanguage(language, context.warnings);
+
+    // Perform version
```

Comment:
>Perform version [](start = 7, length = 15)

[nit] Enforce version? Perform version validation?

Reply:
oops! :)

---
In reply to: [234325149](https://github.com/Microsoft/AdaptiveCards/pull/2171#discussion_r234325149) [](ancestors = 234325149)